sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c

```diff
@@ -424,20 +439,42 @@ static az_result waiting(az_event_policy* me, az_event event)
  */
 static az_result faulted(az_event_policy* me, az_event event)
 {
-  az_result ret = AZ_ERROR_HFSM_INVALID_STATE;
-  (void)me;
-#ifdef AZ_NO_LOGGING
-  (void)event;
-#endif // AZ_NO_LOGGING
+  az_result ret = AZ_OK;
+  az_mqtt5_rpc_server* this_policy = (az_mqtt5_rpc_server*)me;
 
   if (_az_LOG_SHOULD_WRITE(event.type))
   {
     _az_LOG_WRITE(event.type, AZ_SPAN_FROM_STR("az_rpc_server/faulted"));
   }
+  switch (event.type)
+  {
+    case AZ_HFSM_EVENT_ENTRY:
+    {
+      _az_RETURN_IF_FAILED(_az_mqtt5_connection_api_callback(
+          this_policy->_internal.connection,
+          (az_event){ .type = AZ_HFSM_EVENT_ERROR, .data = NULL }));
+      break;
+    }
+    default:
+      ret = AZ_ERROR_HFSM_INVALID_STATE;
+      break;
+  }
 
   return ret;
 }
 
+AZ_NODISCARD az_result az_mqtt5_rpc_server_unsubscribe_begin(az_mqtt5_rpc_server* client)
+{
+  if (client->_internal.connection == NULL)
+  {
+    // This API should be called only when the client is attached to a connection object.
+    return AZ_ERROR_NOT_SUPPORTED;
+  }
+  return _az_hfsm_send_event(
```

Comment:
I think we should be using `az_event_policy_send_outbound` here rather than `az_hfsm_send_event`

Reply:
This is how we're doing it for the other api calls right now. Send_outbound only sends it to the outbound, it doesn't send it to this policy

Reply:
I was thinking of something similar to what is done for `az_mqtt5_rpc_server_register` so moving the preparation of the package to the function for unsubscribing

Reply:
oh, I see! And I think it makes sense to be able to unsubscribe from faulted, which would be the only other limitation this would cause